src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.flink.table.connector.format.DecodingFormat;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.connector.source.ScanTableSource;
+import org.apache.flink.table.connector.source.ScanTableSource.ScanRuntimeProvider;
+import org.apache.flink.table.connector.source.SourceFunctionProvider;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.types.DataType;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class BigQueryDynamicTableSourceTest {
+
+  static String bigqueryReadTable;
+  static ObjectIdentifier tableIdentifier;
+  static TableSchema tableSchema;
+  static Configuration options = new Configuration();
+
+  @BeforeClass
+  public static void setup() {
+    bigqueryReadTable = "q-gcp-6750-pso-gs-flink-22-01.wordcount_dataset.wordcount_output";
+  }
+
+  private DynamicTableSource getTableSource(String tableName) throws Exception {
+
+    MockDynamicTableContext contextObj = createContextObject();
+
+    BigQueryDynamicTableFactory dynamicTableFactory = new BigQueryDynamicTableFactory();
+    BigQueryDynamicTableSource dynamicTableSource =
+        (BigQueryDynamicTableSource) dynamicTableFactory.createDynamicTableSource(contextObj);
+    return dynamicTableSource;
+  }
+
+  @Test
+  public void getScanRuntimeProviderSuccessfulTest() {
+
+    // Mock the big query client factory
+    BigQueryClientFactory mockBigQueryClientFactory = Mockito.mock(BigQueryClientFactory.class);
+
+    ArrowFormatFactory arrowFormat = new ArrowFormatFactory();
+    DataType producedDataType =
+        createContextObject().getCatalogTable().getSchema().toPhysicalRowDataType();
+
+    DecodingFormat<DeserializationSchema<RowData>> decodingFormat =
+        arrowFormat.createDecodingFormat(createContextObject(), options);
+
+    ArrayList<String> readStreamNames = new ArrayList<String>();
+
+    // Initialize the constructor
+    BigQueryDynamicTableSource bigQueryDynamicTableSource =
+        new BigQueryDynamicTableSource(
+            decodingFormat, producedDataType, readStreamNames, mockBigQueryClientFactory);
+
+    ScanRuntimeProvider resultScanRuntimeProvider =
+        bigQueryDynamicTableSource.getScanRuntimeProvider(new MockScanContext());
+
+    assertThat(bigQueryDynamicTableSource instanceof BigQueryDynamicTableSource);
+    assertThat(bigQueryDynamicTableSource.getChangelogMode()).isNotNull();
+    assertThat(bigQueryDynamicTableSource.getChangelogMode() instanceof ChangelogMode);
+    assertThat(bigQueryDynamicTableSource.copy() instanceof BigQueryDynamicTableSource);
+    assertThat(bigQueryDynamicTableSource.asSummaryString()).isEqualTo("BigQuery Table Source");
+    assertThat(resultScanRuntimeProvider).isNotNull();
+  }
+
+  @Test
+  public void testSourceFunction() throws Exception {
```

Comment:
Why are we testing the source function here?

Reply:
removed as not needed here